### ISSUE TITLE
[Enhancement]: myaccount/form-lost-password.php. Normalize actions naming

### ISSUE THREAD

**Issue by dmitrijCraq** [woothemes/woocommerce#36597]
### Describe the solution you'd like

All actions in myaccount/form-lost-password.php contains the word 'lost_password' except for line 34 where the action is called 'woocommerce_lostpassword_form' instead of 'woocommerce_lost_password_form'.

This is the link to the line https://github.com/woocommerce/woocommerce/blob/a531f83b6db32e1d8336e993429c2843c3e32b2c/plugins/woocommerce/templates/myaccount/form-lost-password.php#L34

I'm sure that this is a typo.

### Describe alternatives you've considered

Probably the best way to keep the backwards compatibility with the 'wrong' action name is to deprecate it in favour of the new action called 'woocommerce_lost_password_form' called immediately before or immidiately after the current one.

**Comment by barryhughes** [comment]
> I'm sure that this is a typo.

Yep, probably accidental. I agree re backwards compatibility—we could not for instance simply remove it altogether (or not without a notice period). 

We'd certainly consider any pull request that addresses this, if the status quo is problematic for anyone, though it may need to be supported by a dev advisory or similar before we can ship a change.

**Comment by pierorocca** [comment]
@ralucaStan in your opinion is this a bug, an enhancement, or tech debt?

**Comment by ralucaStan** [comment]
Replaced the label with technical debt

**Comment by samrat-biswas-git** [comment]
Hi @dmitrijCraq - thanks for the report.

I know this has been open for a very long time, and I appreciate your patience. We're currently going through some older WooCommerce issues, identifying what still needs fixing, and tidying up what's already fixed or out of scope.

I've taken a look and marked this ready for the engineering team to pick up.

What I'm seeing: The lost-password template still only fires `woocommerce_lostpassword_form`; there is no normalized `woocommerce_lost_password_form` companion hook on trunk. The issue thread already captures the expected backwards-compatible approach of adding a new hook while keeping the old one.
